File: 백준/Silver/28117. prlong longf/prlong longf.py
N = int(input())
s = input()

s = list(s.replace("long", ".").strip())
N = len(s)
dp = [0 for _ in range(100)]
dp[0] = dp[1] = 1
for i in range(2, 100):
    dp[i] = dp[i-1]+dp[i-2]

ans = 1
tmp = []
i = 0
while i < N:
    # print(tmp, ans)
    if s[i] == '.':
        tmp.append(s[i])
        i += 1
        continue
    else:
        # print(tmp)
        if len(tmp) > 0:
            ans *= dp[len(tmp)]
        tmp = []
    i += 1
if len(tmp) > 0:
    ans *= dp[len(tmp)]
print(ans)

File: 백준/Silver/28117. prlong longf/prlong longf.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String args[]) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        // StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(br.readLine());
        String s = br.readLine().replace("long",".");
        N = s.length();

        int[] dp = new int[100];
        dp[0]=dp[1]=1;
        for (int i=2;i<100;i++)
            dp[i] = dp[i-1]+dp[i-2];

        int i=0;
        int ans=1;
        while (i<N){
            if (s.charAt(i)=='.'){
                List<Character> tmp = new ArrayList<>();
                while (i<N && s.charAt(i)=='.'){
                    tmp.add('.');
                    i++;
                }
                ans*=dp[tmp.size()];
            }
            i++;
        }
        System.out.println(ans);
    }
}